shopfloor_reception/services/reception.py

```diff
@@ -888,23 +907,21 @@ def set_destination(
         search = self._actions_for("search")
         location = search.location_from_scan(location_name)
         move_dest_location = selected_line.location_dest_id
-        move_child_locations = self.env["stock.location"].search(
-            [("id", "child_of", move_dest_location.id), ("usage", "!=", "view")]
-        )
         pick_type_dest_location = picking.picking_type_id.default_location_dest_id
-        pick_type_child_locations = self.env["stock.location"].search(
-            [("id", "child_of", pick_type_dest_location.id), ("usage", "!=", "view")]
-        )
-        if location not in move_child_locations | pick_type_child_locations:
+        # TODO: Extract in different method, use everywhere to check locations
+        if location.usage == "view" or not (
+            move_dest_location.parent_path.startswith(location.parent_path)
+            or pick_type_dest_location.parent_path.startswith(location.parent_path)
+        ):
             return self._response_for_set_destination(
                 picking,
                 selected_line,
                 message=self.msg_store.dest_location_not_allowed(),
             )
-        if location in move_child_locations:
+        if move_dest_location.parent_path.startswith(location.parent_path):
```

Comment:
this is the other way around.
We want to check that location is a child of move_dest_location
```"a-b-c".startswith("a-b")```
In your case, you check that the selected location is a parent of the one set on the move.
To me those string methods doesn't makes things easier to understand.
I'd rather stick to the `child_of` search operator.

Reply:
Also, parent_path is a simple field, which can be changed by hand by any user with the rights to do so.
I really wouldn't rely on this.

Reply:
I'm following the indications by @jbaudoux . It does make sense that it's the other way round, as you say. 

@jbaudoux thoughts?

Reply:
indeed should have been the other way around

Reply:
And it doesn't seems right to me, as this is a regular char field that might be edited.
Doesn't look safe to me.

Reply:
As discussed, `child_of` itself uses `parent_path` so we can use the latter to simplify the check.

I've pushed a fixup commit to invert the location check as well.